### DISCUSSION TITLE
修改JSON文件后，插件无法工作

### DISCUSSION THREAD

**Discussion by gdxia** [DominikDoom/a1111-sd-webui-tagcomplete#183] [General]
比如我改成这样：

{
    "danbooru": {
        "-1": ["red", "maroon"],
        "0": ["lightblue", "dodgerblue"],
        "1": ["indianred", "firebrick"],
        "3": ["violet", "darkorchid"],
        "4": ["lightgreen", "darkgreen"],
        "5": ["orange", "darkorange"]
    },
    "e621": {
        "-1": ["red", "maroon"],
        "0": ["lightblue", "dodgerblue"],
        "1": ["gold", "goldenrod"],
        "3": ["violet", "darkorchid"],
        "4": ["lightgreen", "darkgreen"],
        "5": ["tomato", "darksalmon"],
        "6": ["red", "maroon"],
        "7": ["whitesmoke", "black"],
        "8": ["seagreen", "darkseagreen"]
    }
 "extra-quality-tags": {
    "0": ["lightblue", "dodgerblue"],
    "1": ["gold", "goldenrod"],
    "2": ["violet", "darkorchid"],
    "3": ["lightgreen", "darkgreen"]
   }
}

**Comment by DominikDoom** [comment]
如果这正是你使用的文本，有两件事：

1. 第二组后面少了一个逗号，这就是为什么文件不能被解析的原因，因为它是无效的JSON。
2. 即使如此，质量标签或一般的额外标签也不打算作为它们自己的组来使用。相反，它们使用你的主标签文件中的类别颜色。因此，除非你的主标签文件也被称为`extra-quality-tags`，否则它不会有任何影响。不过，如果你想要的话，你可以在现有的组中添加新的颜色，或者重新命名你的标签文件以匹配新的自定义组。例如，给你的质量标签颜色的类别编号为6-9，并将它们添加到danbooru组。

**Reply by gdxia** [reply]
太感谢了，大神呀，我现在懂了，并操作成功了。

1. 原来是我的JSON语法错误, 组与组之前要加逗号的。
2. 附加的文件不需要定义标签颜色，附加文件的颜色使用的是主文件的颜色定义。

这个插件非常好，我今天把这个插件的所以设置项都体验了一遍。如果充分利用这个插件，可以大大提升自己的工作效率。非常感谢。
